lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -390,14 +432,19 @@ def parse(action, xml)
       end
 
       def parse_element(raw, node)
+        node_name = node.name.underscore
         node.attributes.each do |k, v|
-          raw["#{node.name.underscore}_#{k.underscore}".to_sym] = v
+          raw["#{node_name}_#{k.underscore}".to_sym] = v
         end
         if node.has_elements?
-          raw[node.name.underscore.to_sym] = true unless node.name.blank?
+          raw[node_name.to_sym] = true unless node.name.blank?
           node.elements.each { |e| parse_element(raw, e) }
         else
-          raw[node.name.underscore.to_sym] = node.text unless node.text.nil?
```

Comment:
This allows us to include the error message when there are newlines. I discovered this issue when trying to make assertions about the returned error message from existing canned responses in the unit tests. Now we can return the error message whether the `CDATA` information is preceded and/or followed by newlines or not.